src/app/root.tsx

```diff
@@ -0,0 +1,115 @@
+
+import { cryptoWaitReady } from '@polkadot/util-crypto';
+
+import { $path } from 'remix-routes';
+
+import { BalanceProvider } from '@/common/balances';
+import { ChainRegistry } from '@/common/chainRegistry';
+import { ExtrinsicProvider } from '@/common/extrinsicService';
+import { GlobalStateProvider, useGlobalContext } from '@/common/providers/contextProvider.tsx';
+import { TelegramProvider } from '@/common/providers/telegramProvider.tsx';
+import { getWallet } from '@/common/wallet';
+import { ErrorScreen, LoadingScreen } from '@/components';
+
+import stylesheet from './tailwind.css?url';
+
+export const links: LinksFunction = () => [
+  { rel: 'stylesheet', href: stylesheet },
+  { rel: 'shortcut icon', href: 'data:image/x-icon;,', type: 'image/x-icon' },
+];
+
+export const meta: MetaFunction = () => [
+  { charSet: 'utf-8' },
+  {
+    name: 'viewport',
+    content: 'width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no',
+  },
+];
+
+export const ErrorBoundary: FC = () => {
+  const error = useRouteError();
+
+  return <ErrorScreen error={error instanceof Error ? error.toString() : 'Unknown error'} />;
+};
+
+export const HydrationFallback: FC = () => <LoadingScreen />;
+
+export const Layout: FC<PropsWithChildren> = ({ children }) => (
+  <html lang="en">
+    <head>
+      <Meta />
+      <Links />
+
+      <link rel="preconnect" href="https://fonts.googleapis.com" />
+      <link rel="preconnect" href="https://fonts.gstatic.com" crossOrigin="anonymous" />
+      <link href="https://fonts.googleapis.com/css2?family=Manrope:wght@200..800&display=swap" rel="stylesheet" />
+    </head>
+    <body>
+      <DataContext>{children}</DataContext>
+      <ScrollRestoration />
+      <script defer src="https://telegram.org/js/telegram-web-app.js" />
+      <script defer src="https://widget.mercuryo.io/embed.2.0.js" />
+      <Scripts />
+    </body>
+  </html>
+);
+
+const DataContext: FC<PropsWithChildren> = ({ children }) => {
+  const [loading, setLoading] = useState(true);
+  useEffect(() => {
+    cryptoWaitReady().then(() => setLoading(false));
+  }, []);
+
+  return (
+    <GlobalStateProvider>
+      {loading ? (
+        <LoadingScreen />
+      ) : (
+        <TelegramProvider>
+          <ChainRegistry>
+            <ExtrinsicProvider>
+              <BalanceProvider>{children}</BalanceProvider>
+            </ExtrinsicProvider>
+          </ChainRegistry>
+        </TelegramProvider>
+      )}
+    </GlobalStateProvider>
+  );
+};
+
+export default function App() {
+  const [error, setError] = useState<string | null>(null);
+  const { setPublicKey } = useGlobalContext();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    getWallet()
+      .then(wallet => {
+        if (!wallet) {
+          return navigate($path('/onboarding'), { replace: true });
+        }
+        setPublicKey(wallet?.publicKey);
+        navigate($path('/dashboard'), { replace: true });
```

Comment:
This `$path` function is part of `remix-routes` lib. It generate types based on routes in file system, imo it's better solution because it has
* automatic type update
* `navigate($path('/settings/change-password/'))` looks more readable for me, than `navigate(Paths.SETTINGS_CHANGE_PASSWORD)`, idk